specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,214 @@
+
+The OP Stack Manager is a contract that deploys the L1 contracts for an OP Stack chain in a single
+transaction. It provides a minimal set of user-configurable parameters to ensure that the resulting
+chain meets the [standard configuration] requirements.
+
+The version deployed is always a governance-approved contract release. The set
+of governance approved [contract releases] can be found on the
+[Optimism Monorepo releases] page, and is the set of releases named
+`op-contracts/vX.Y.Z`.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Deployment](#deployment)
+- [Interface](#interface)
+  - [`Proxy.sol`](#proxysol)
+  - [`deploy`](#deploy)
+  - [Getter Methods](#getter-methods)
+- [Implementation](#implementation)
+  - [Batch Inbox Address](#batch-inbox-address)
+  - [Contract Deployments](#contract-deployments)
+- [Security Considerations](#security-considerations)
+  - [Chain ID Source of Truth](#chain-id-source-of-truth)
+  - [Chain ID Frontrunning](#chain-id-frontrunning)
+  - [Chain ID Value](#chain-id-value)
+  - [Proxy Admin Owner](#proxy-admin-owner)
+  - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deployment
+
+The OP Stack Manager is a proxied contract deployed at `0xTODO`. It can be deployed as follows:
+
+TODO.
+
+## Interface
+
+Version 1.0.0 of the OP Stack Manager deploys the [`op-contracts/v1.4.0`]
+contracts release.
+
+### `Proxy.sol`
+
+The OP Stack Manager is a proxied contract using the standard `Proxy.sol` contract that lives in
+the Optimism monorepo. Therefore the OP Stack Manager will have the same interface as the
+`Proxy.sol`, in addition to other methods defined in this specification.
+
+The privileged methods of the OP Stack Manager will be held by the L1 ProxyAdmin owner, as
+specified by the [standard configuration].
+
+### `deploy`
+
+The `deploy` method is the only non-view method in the contract. It is used to
+deploy the full set of L1 contracts required to setup a new OP Stack chain that
+complies with the [standard configuration]. It has the following interface:
+
+```solidity
+struct Roles {
+  address proxyAdminOwner;
+  address systemConfigOwner;
+  bytes32 batcherHash;
+  address unsafeBlockSigner;
+  address proposer;
+  address challenger;
+}
+
+function deploy(
+  uint64 l2ChainId,
+  Roles roles,
+  uint256 scalar
+) external returns (SystemConfig)
+```
+
+The `l2ChainId` has the following restrictions:
+
+- It must not be equal to 0.
+- It must not be equal to the chain ID of the chain the OP Stack Manager is
+deployed on.
+- It must not be equal to a chain ID that is already present in the
+[ethereum-lists/chains] repository. This is not enforced onchain, but may matter
+for future versions of OP Stack Manager that handle upgrades.
```

Comment:
Part of the standard config checks include chain ID uniqueness checks against that repo. I think we should keep this bullet, instead of (adding or replacing with) a "must not be a chain id of a chain previously deployed using the op stack manager", because that would e.g. allow a chain ID of 10 since OP Mainnet wasn't deployed through OPSM.

In other words, chain ID uniquess for standard chains has to be enforced offchain anyway, so I don't see much value in adding that requirement/check. I've expanded on the logic in be28df1, let me know what you think